skimage/morphology/_skeletonize.py

```diff
@@ -249,22 +248,23 @@ def thin(image, max_iter=None):
            [0, 0, 0, 0, 0, 0, 0],
            [0, 0, 0, 0, 0, 0, 0]], dtype=uint8)
     """
-    # check parameters
-    max_iter = max_iter or sys.maxsize
     # check that image is 2d
     assert_nD(image, 2)
 
     # convert image to uint8 with values in {0, 1}
     skel = np.asanyarray(image, dtype=bool).astype(np.uint8)
 
     # neighborhood mask
-    mask = np.array([[ 8,  4,  2],
-                     [16,  0,  1],
-                     [32, 64,128]], dtype=np.uint8)
+    mask = np.array([[ 8,  4,   2],
+                     [16,  0,   1],
+                     [32, 64, 128]], dtype=np.uint8)
 
     # iterate until convergence, up to the iteration limit
-    for i in range(max_iter):
-        before = np.sum(skel)  # count points before thinning
+    iter_cnt = max_iter or np.inf
+    n_pts_old, n_pts_new = np.inf, np.sum(skel)
+    while not np.isclose(n_pts_old, n_pts_new, rtol=0) and iter_cnt > 0:
```

Comment:
Why is this not `n_pts_old != n_pts_new`?

Reply:
I also think you could count up, `while n_iter < max_iter`. This would be more readable, and I believe it would still work?

Reply:
@jni Done, thanks!